dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java

```diff
@@ -225,102 +230,110 @@ default List<SourceFile> getAllSecondaryFiles(long workflowId, String tag, Descr
      * @param fileType the filetype we want to consider
      * @return a list of SourceFile
      */
-    default List<SourceFile> getAllSourceFiles(long workflowId, String tag, DescriptorLanguage.FileType fileType, Optional<User> user, FileDAO fileDAO) {
-        final Map<String, ImmutablePair<SourceFile, FileDescription>> sourceFiles = this.getSourceFiles(workflowId, tag, fileType, user, fileDAO);
+    default List<SourceFile> getAllSourceFiles(long workflowId, String tag, DescriptorLanguage.FileType fileType, Optional<User> user, FileDAO fileDAO, VersionDAO versionDAO) {
+        final Map<String, ImmutablePair<SourceFile, FileDescription>> sourceFiles = this.getSourceFiles(workflowId, tag, fileType, user, fileDAO,
+                versionDAO);
         return sourceFiles.entrySet().stream().filter(entry -> entry.getValue().getLeft().getType().equals(fileType))
             .map(entry -> entry.getValue().getLeft()).collect(Collectors.toList());
     }
 
     /**
      * This returns a map of file paths -> pairs of sourcefiles and descriptions of those sourcefiles
+     *
      * @param workflowId the database id for a workflow
-     * @param tag the version of the workflow
-     * @param fileType the type of file we're interested in
+     * @param tag        the version of the workflow
+     * @param fileType   the type of file we're interested in
+     * @param versionDAO
      * @return a map of file paths -> pairs of sourcefiles and descriptions of those sourcefiles
      */
     default Map<String, ImmutablePair<SourceFile, FileDescription>> getSourceFiles(long workflowId, String tag,
-            DescriptorLanguage.FileType fileType, Optional<User> user, FileDAO fileDAO) {
-        T entry = getDAO().findById(workflowId);
-        checkOptionalAuthRead(user, entry);
-
-        // tighten permissions for hosted tools and workflows
-        if (!user.isPresent() || AuthenticatedResourceInterface.userCannotRead(user.get(), entry)) {
-            if (!entry.getIsPublished()) {
-                if (entry instanceof Tool && ((Tool)entry).getMode() == ToolMode.HOSTED) {
-                    throw new CustomWebApplicationException("Entry not published", HttpStatus.SC_FORBIDDEN);
-                }
-                if (entry instanceof Workflow && ((Workflow)entry).getMode() == WorkflowMode.HOSTED) {
-                    throw new CustomWebApplicationException("Entry not published", HttpStatus.SC_FORBIDDEN);
+            DescriptorLanguage.FileType fileType, Optional<User> user, FileDAO fileDAO, VersionDAO versionDAO) {
+        try {
+            versionDAO.enableNameFilter(tag);
```

Comment:
Later (currently at PR line 272), we set `tag` to `latest` if it is null.  For the code to work, the value we check for there needs to match the value we set here.  Otherwise, in the `tag` is null case, the version we're looking for will be filtered out before we search for it.  Also, I'm not sure what happens when you enable a filter and and set a parameter value to null, but it might not be good...

Reply:
Ah, good catch! Embarrassed I missed that.

Reply:
No worries! :)